numpy/ma/core.py

```diff
@@ -3802,7 +3802,10 @@ def __iadd__(self, other):
         else:
             if m is not nomask:
                 self._mask += m
-        ndarray.__iadd__(self._data, np.where(self._mask, 0, getdata(other)))
+        ndarray.__iadd__(
```

Comment:
As it was already using `ndarray,__iadd__` (or equivalent), I thought I would leave it as I figured there had to be a good reason it was here at all. This seems to be true of a number of other methods in `MaskedArray`. Should we really be changing this?